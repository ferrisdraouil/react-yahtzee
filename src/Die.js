import React, { Component } from 'react';
import './Die.css';

class Die extends Component {
  render() {
    return (
      <button
        className={this.props.locked ? 'Die Die-locked' : 'Die'}
        onClick={() => this.props.handleClick(this.props.idx)}
      >
        // can not be clicked if the this.props.locked is true
        {this.props.val}
      </button>
    );
  }
}

export default Die;
