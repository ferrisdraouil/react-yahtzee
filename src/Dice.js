import React, { Component } from 'react';
import Die from './Die';
import './Dice.css';

class Dice extends Component {
  render() {
    return (
      <div className="Dice">
        {this.props.dice.map((d, idx) => (
          <Die
            handleClick={this.props.handleClick}
            val={d}
            locked={this.props.locked[idx]}
            idx={idx}
            key={idx}
          />
        ))}
      </div>
    );
  }
}

Dice.defaultProps = {
  //dice: Array.from({ length: 5 }, () => Math.ceil(Math.random() * 6))
  dice: [1, 2, 3, 4, 5]
};

export default Dice;
