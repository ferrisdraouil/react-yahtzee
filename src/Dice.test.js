import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dice from './Dice';
import Die from './Die';

it('renders without crashing', function() {
  shallow(<Dice dice={Array(5).fill(false)} locked={Array(5).fill(false)} />);
});
it('renders without crashing with mount', function() {
  mount(<Dice dice={Array(5).fill(false)} locked={Array(5).fill(false)} />);
});

it('matches unlocked snapshot', function() {
  let wrapper = mount(
    <Dice
      dice={Array(5).fill(false)}
      locked={[true, false, true, false, true]}
    />
  );
  let dieOneProps = wrapper
    .find(Die)
    .at(0)
    .props();
  let dieTwoProps = wrapper
    .find(Die)
    .at(1)
    .props();
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
  // expect(dieOneProps.locked).toEqual(true);
  // expect(dieTwoProps.locked).toEqual(false);
  wrapper.unmount();
});
