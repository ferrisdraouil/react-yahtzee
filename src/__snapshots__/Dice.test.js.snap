// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches unlocked snapshot 1`] = `
<Dice
  dice={
    Array [
      false,
      false,
      false,
      false,
      false,
    ]
  }
  locked={
    Array [
      true,
      false,
      true,
      false,
      true,
    ]
  }
>
  <div
    className="Dice"
  >
    <Die
      idx={0}
      key="0"
      locked={true}
      val={false}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      />
    </Die>
    <Die
      idx={1}
      key="1"
      locked={false}
      val={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
    <Die
      idx={2}
      key="2"
      locked={true}
      val={false}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      />
    </Die>
    <Die
      idx={3}
      key="3"
      locked={false}
      val={false}
    >
      <button
        className="Die"
        onClick={[Function]}
      />
    </Die>
    <Die
      idx={4}
      key="4"
      locked={true}
      val={false}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      />
    </Die>
  </div>
</Dice>
`;
